### PR TITLE
[IMP]website, website_editor: changes so that pictures can be saved a…

### PR DESCRIPTION
**Pull request by pgu-odoo**
…ccording to language on website.

Description of the issue/feature this PR addresses:
https://www.odoo.com/web#id=33372&view_type=form&model=project.task&action=327&menu_id=5200
Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: odoo/addons/base/ir/ir_translation.py [hunk 1]
```diff
@@ -469,7 +469,7 @@ def _sync_terms_translations(self, field, records):
                 if trans.src == trans.value:
                     discarded += trans
                 elif trans.src not in terms:
-                    matches = get_close_matches(trans.src, terms, 1, 0.9)
+                    matches = get_close_matches(trans.src, terms, 1, 1)
```
Comment (rco-odoo) [inline odoo/addons/base/ir/ir_translation.py]: This screws up a functionality: it allows to not lose translations when you fix a typo.
This is clearly wrong.

File: odoo/addons/base/ir/ir_translation.py [hunk 2]
```diff
@@ -544,8 +544,8 @@ def _check_value(self):
                 mname, fname = trans.name.split(',')
                 record = trans.env[mname].browse(trans.res_id)
                 field = record._fields[fname]
-                if callable(field.translate):
-                    src = trans.src
+                src = trans.src
+                if callable(field.translate) and not src.startswith('<img'):
```
Comment (rco-odoo) [inline odoo/addons/base/ir/ir_translation.py]: The element `img` is translatable, but that does not mean is always appear at the beginning of the source term. Consider:
```html
    <div>Please look at <img src="..."></div>
```
The source term to translate will be `'Please look at '`.

Comment (pgu-odoo) [reply odoo/addons/base/ir/ir_translation.py]: I have checked all snippets and no where text is using like this. Everywhere if text is used, it is closed in <p> tag outside div of image tag. like -
               <p>Please look at </p>
               <div><div>

Although, changed the code to find out the image tag. Now using regular expression for this.

**Comment by jat-odoo** [comment]
LGTM

**Comment by rdeodoo** [comment]
According to the task, this was replaced by https://github.com/odoo/odoo/pull/33436
Can we close?

**Pull request closed without merging**